api/src/main/java/org/eclipse/microprofile/config/spi/ConfigSource.java

```diff
@@ -84,7 +84,8 @@
     Map<String, String> getProperties();
 
     /**
-     * Gets all property names known to this config source, without evaluating the values.
+     * Gets all property names known to this config source, without evaluating the values. 
+     * The returned property names may be a subset of the configurations in this config source, due to security or performance reasons.
```

Comment:
```suggestion
     * The returned property names may be a subset of the names of the total set of retrievable properties in this config source.
```

I don't think we want to constrain the purposes for which the names can be a subset.

Reply:
ok. @dmlloyd I have updated the javadoc as you recommended. Please check.